server-timing/load.php

```diff
@@ -113,3 +116,66 @@ function perflab_wrap_server_timing( $callback, $metric_slug, $access_cap ) {
 		return $result;
 	};
 }
+
+/**
+ * Registers the Server-Timing setting.
+ *
+ * @since n.e.x.t
+ */
+function perflab_register_server_timing_setting() {
+	register_setting(
+		PERFLAB_SERVER_TIMING_SCREEN,
+		PERFLAB_SERVER_TIMING_SETTING,
+		array(
+			'type'              => 'object',
+			'sanitize_callback' => 'perflab_sanitize_server_timing_setting',
+			'default'           => array(),
+		)
+	);
+}
+add_action( 'init', 'perflab_register_server_timing_setting' );
+
+/**
+ * Sanitizes the Server-Timing setting.
+ *
+ * @since n.e.x.t
+ *
+ * @param mixed $value Server-Timing setting value.
+ * @return array Sanitized Server-Timing setting value.
+ */
+function perflab_sanitize_server_timing_setting( $value ) {
+	if ( ! is_array( $value ) ) {
+		return array();
+	}
+
+	/*
+	 * Ensure that every element is an indexed array of hook names.
+	 * Any duplicates across a group of hooks are removed.
+	 */
+	return array_filter(
+		array_map(
+			static function( $hooks ) {
+				if ( ! is_array( $hooks ) ) {
+					$hooks = explode( "\n", $hooks );
+				}
+				return array_values(
+					array_unique(
+						array_filter(
+							array_map(
+								static function( $hookname ) {
+									return preg_replace(
+										'/[^A-Za-z0-9_\-\/\.]/',
+										'',
+										strtolower( $hookname )
+									);
+								},
+								$hooks
+							)
```

Comment:
Would this be simpler and less prone to munge complex hook names?

```suggestion
							array_map( 'sanitize_text_field', $hooks )
```

(Note: It also does `trim()` automatically.)

Reply:
`sanitize_text_field` allows spaces, which I really don't think we should allow here. How about we use `preg_replace( '/\s/', '', sanitize_text_field( $value ) )`?

Reply:
Spaces could appear in hooks though, right? Humm, but it seems they're mostly typos: https://wpdirectory.net/search/01H78JM9J0FXXMR2BS0JC4T0TN

So I'm fine with stripping all whitespace from the hook names.

Reply:
Updated in https://github.com/WordPress/performance/pull/784/commits/b85959e844956d00997748d7b2ee321b71364c8d